### PR TITLE
Try cleanup DPS custom allocator function, and add other solution files

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: azureiot.sln [hunk 1]
```diff
@@ -175,6 +173,37 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SimulatedDevice", "iothub\d
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ReadD2cMessages", "iothub\service\samples\getting started\ReadD2cMessages\ReadD2cMessages.csproj", "{F4DA6FA4-7655-4116-994B-9B2E3D3522C0}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "vsts", "vsts", "{CC532097-89B3-4971-97CB-A790573E38DA}"
```
Comment (drwill-ms) [inline azureiot.sln:176]: Adding more things we should readily access in VS in a solution folder.

File: e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/function.proj [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
 
-    <ItemGroup> 
-        <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Service" Version="1.5.1" />  
-        <PackageReference Include="Microsoft.Azure.Devices.Shared" Version="1.16.1" />  
-    </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Service" Version="1.18.2" />
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/function.proj:7]: Update to latest v1 nuget packages.

Someday, we'll update this to v2.

File: e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/run.csx [hunk 3]
```diff
@@ -1,85 +1,92 @@
...
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.Devices.Provisioning.Service;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json;
 
-//This function will return the iot hub hostname to provision to based on which of the list of hub names has the longest host name
-public static async Task<HttpResponseMessage> Run(HttpRequestMessage req, TraceWriter log)
+// This function will return the iot hub hostname to provision to based on which of the list of hub names has the longest host name.
+public static async Task<IActionResult> Run(HttpRequestMessage req, ILogger log)
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/run.csx:12]: We had a warning that we were using a logger that will be deprecated.

File: e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/run.csx [hunk 4]
```diff
@@ -1,85 +1,92 @@
...
-                Content = new StringContent(
-                    JsonConvert.SerializeObject(obj, Formatting.Indented),
-                    Encoding.UTF8,
-                    "application/json")
-            };
-}    
+        string responsePayload = JsonConvert.SerializeObject(obj, Formatting.Indented);
+        log.LogInformation($"Response payload is: [{responsePayload}].");
+
+        return (fail)
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/DpsCustomAllocatorFunctionFiles/run.csx:76]: New kinds of return values with the new method signature.

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 5]
```diff
@@ -224,12 +224,8 @@ $iotHubX509DevicePfxPath = "$PSScriptRoot/IotHubX509Device.pfx";
 $iotHubX509CertChainDeviceCommonName = "Save_iothubx509chaindevice1";
 $iotHubX509ChainDevicPfxPath = "$PSScriptRoot/IotHubX509ChainDevice.pfx";
 
-$dpsX509GroupEnrollmentDeviceCertCommonName = "xdevice1";
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:227]: Unused.

File: e2e/test/prerequisites/run.csx [hunk 6]
```diff
@@ -1,85 +0,0 @@
-#r "Newtonsoft.Json"
```
Comment (drwill-ms) [inline e2e/test/prerequisites/run.csx:1]: Dupe file

File: e2e/test/provisioning/ProvisioningE2ETests.cs [hunk 7]
```diff
@@ -889,10 +889,9 @@ public async Task DPS_Registration_AmqpWs_X509_IndividualEnrollment_InvalidGloba
             var customAllocationDefinition = new CustomAllocationDefinition
             {
                 WebhookUrl = TestConfiguration.Provisioning.CustomAllocationPolicyWebhook,
-                ApiVersion = "2019-03-31",
+                ApiVersion = "2021-10-01",
```
Comment (drwill-ms) [inline e2e/test/provisioning/ProvisioningE2ETests.cs:892]: Updating to latest data plane API version.

File: provisioning/service/src/Contract/SDKUtils.cs [hunk 8]
```diff
@@ -5,7 +5,7 @@ namespace Microsoft.Azure.Devices.Provisioning.Service
 {
     internal class SdkUtils
     {
-        private const string ApiVersionProvisioning = "2019-03-31";
-        public const string ApiVersionQueryString = "api-version=" + ApiVersionProvisioning;
+        private const string ApiVersionProvisioning = "2021-10-01";
```
Comment (drwill-ms) [inline provisioning/service/src/Contract/SDKUtils.cs:8]: latest API version

**Comment by timtay-microsoft** [comment]
FYI, these custom allocation policy tests may fall under "testing the service" more than "testing the SDK". From the DPS device SDK's perspective, the custom allocation policy is a service implementation detail. We may be able to get by on a single test wherein the service client creates an enrollment with a custom allocation policy, but no device is ever provisioned from it. We'd still need a custom allocation function deployed, but we'd be less susceptible to issues like this in the future because the function would just return a random IoT hub.

**Review by timtay-microsoft — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]
No issue with the service then?

**Comment by drwill-ms** [comment]
Failures are max quota again, unrelated to this PR's changes and passed before. Merging.

**Pull request merged**